controller/attribute/manager.py

```diff
@@ -1,42 +1,111 @@
-from typing import List
+from typing import List, Tuple
 from controller.tokenization.tokenization_service import request_tokenize_project
-from submodules.model.business_objects import attribute, general
+from submodules.model.business_objects import attribute, record, tokenization
 from submodules.model.models import Attribute
-from util import daemon
+from submodules.model.enums import AttributeState, DataTypes
+from util import daemon, notification
+
+from . import util
 
 
 def get_attribute(project_id: str, attribute_id: str) -> Attribute:
     return attribute.get(project_id, attribute_id)
 
 
-def get_all_attributes(project_id: str) -> List[Attribute]:
-    return attribute.get_all_ordered(project_id, True)
+def get_all_attributes(
+    project_id: str, state_filter: List[str] = None
+) -> List[Attribute]:
+    return attribute.get_all_ordered(project_id, True, state_filter)
 
 
 def check_composite_key(project_id: str) -> bool:
     return attribute.check_composite_key_is_valid(project_id)
 
 
-def create_attribute(project_id: str, name: str) -> None:
+def create_attribute(project_id: str, name: str) -> Attribute:
     relative_position: int = attribute.get_relative_position(project_id)
     if relative_position is None:
         relative_position = 1
     else:
         relative_position += 1
 
-    attribute.create(project_id, name, relative_position, with_commit=True)
+    attribute_item: Attribute = attribute.create(
+        project_id,
+        name,
+        relative_position,
+        with_commit=True,
+    )
+    return attribute_item
+
+
+def create_user_attribute(project_id: str) -> Attribute:
+    relative_position: int = attribute.get_relative_position(project_id)
+    if relative_position is None:
+        relative_position = 1
```

Comment:
just personal opinion: instead of overwriting the `relative_position` that you found in the db, you could also say `relative_position_new` or `relative_position_added`, or `relative_position_increment` (you get what i mean). As you add 1 to the existing value

Reply:
changed